tests/utils_tests/test_functional.py

```diff
@@ -47,43 +49,107 @@ def __bytes__(self):
         self.assertEqual(str(t), "Î am ā Ǩlâzz.")
         self.assertEqual(bytes(t), b"\xc3\x8e am \xc4\x81 binary \xc7\xa8l\xc3\xa2zz.")
 
+    def assertCachedPropertyWorks(self, attr, Class):
+        with self.subTest(attr=attr):
+            def get(source):
+                return getattr(source, attr)
```

Comment:
I feel like this _helper_ is making the assertion harder to reason about. Inlining `getattr(source, attr)` calls would be easier to read IMO.

Reply:
it gets very long winded when you add `getattr(source, attr)` eveywhere